scripts/pca_variant_filter.py

```diff
@@ -74,313 +72,369 @@ def determine_pca_variants(
-        snv_only=snv_only,
-        bi_allelic_only=bi_allelic_only,
-        min_gnomad_v3_ac=min_gnomad_v3_ac,
-        high_qual_ccdg_exome_interval_only=high_qual_ccdg_exome_interval_only,
-        high_qual_ukbb_exome_interval_only=high_qual_ukbb_exome_interval_only,
-        filter_lcr=filter_lcr,
-        filter_segdup=filter_segdup,
-        min_af=min_joint_af,
-        min_callrate=min_joint_callrate,
-        min_ccdg_exome_callrate=min_ccdg_exome_callrate,
-        min_ukbb_exome_callrate=min_ukbb_exome_callrate,
-        min_inbreeding_coeff_threshold=min_inbreeding_coeff_threshold,
-        min_hardy_weinberg_threshold=min_hardy_weinberg_threshold,
-    )
+        ht = ht.filter(variant_filter_expr)
+
+        ht = ht.annotate_globals(
+            autosomes_only=autosomes_only,
+            snv_only=snv_only,
+            adj_only=adj_only,
+            bi_allelic_only=bi_allelic_only,
+            min_gnomad_v3_ac=min_gnomad_v3_ac,
+            high_qual_ccdg_exome_interval_only=high_qual_ccdg_exome_interval_only,
+            high_qual_ukbb_exome_interval_only=high_qual_ukbb_exome_interval_only,
+            filter_lcr=filter_lcr,
+            filter_segdup=filter_segdup,
+            min_af=min_joint_af,
+            min_callrate=min_joint_callrate,
+            min_ccdg_exome_callrate=min_ccdg_exome_callrate,
+            min_ukbb_exome_callrate=min_ukbb_exome_callrate,
+            min_inbreeding_coeff_threshold=min_inbreeding_coeff_threshold,
+            min_hardy_weinberg_threshold=min_hardy_weinberg_threshold,
+        )
 
-    ht = filter_low_conf_regions(
-        ht,
-        filter_lcr=filter_lcr,
-        filter_decoy=False,  # No decoy for GRCh38
-        filter_segdup=filter_segdup,
-    )
+        ht = filter_low_conf_regions(
+            ht,
+            filter_lcr=filter_lcr,
+            filter_decoy=False,  # No decoy for GRCh38
+            filter_segdup=filter_segdup,
+        )
 
-    ht = ht.checkpoint(
-        get_pca_variants_path(ld_pruned=False),
-        overwrite=overwrite,
-        _read_if_exists=(not overwrite),
-    )
+        ht = ht.checkpoint(
+            get_pca_variants_path(ld_pruned=False, filter_washu=filter_washu),
+            overwrite=True,
+        )
+    else:
+        ht = hl.read_table(
+            get_pca_variants_path(
+                ld_pruned=False, data=ld_pruning_dataset, filter_washu=filter_washu
+            )
+        )
 
     if ld_pruning:
         # Whether this is still required?
         logger.warning(
             "The LD-prune step of this function requires non-preemptible workers only!"
         )
         logger.info("Creating Table after LD pruning of %s...", ld_pruning_dataset)
-        ht = hl.read_table(get_pca_variants_path(ld_pruned=False))
         if ld_pruning_dataset == "ccdg_genomes":
-            vds = hl.vds.read_vds(get_ccdg_vds_path("genomes"))
-            vds = hl.vds.split_multi(vds)
+            vds = get_ccdg_vds("genomes")
```

Comment:
Just confirming, you don't want washu filtered here if specified by `filter_washu`?

Reply:
No, I didn't filter WashU samples here. Because we previously wanted to do some sample-wise analysis using the pre-ld pruning matrix table. And then I just had that full matrix table go into LD pruning. Does that seem to cause any problem?

Reply:
I think this is completely up to you, below I use the gnomAD v3 VDS for this so it doesn't impact gnomAD, it is very likely not going to impact the final set for you either. I just wanted to confirm the CDDG VDS you wanted since now `filter_washu` is a parameter in `get_ccdg_vds.